runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/PuzzleSolverOverlay.java

```diff
@@ -295,58 +262,12 @@ private void cacheItems(int[] items)
 			}
 		}
 
-		return convertedItems;
-	}
-
-	private BufferedImage getDownArrow()
-	{
-		if (downArrow == null)
+		//If blank is in last position, items doesn't contain it. So we add it manually.
+		if (items.length == PuzzleSolver.SIZE - 1)
```

Comment:
Are you sure that the "items" array is smaller when the blank is in the last position? Seems kind of illogical but maybe that's just how it is?

Reply:
Yes, i found it out because the solver didn't work if you opened the puzzle while the blank was in the last position.I have manually confirmed it aswell.